### PR TITLE
implemented s3 cache handler

### PR DESCRIPTION
**Pull request by RomanBobrovskiy**
- updated generating cache keys
- updated structure and path to store cached values
- excluded images and redirects from getting cached
- added S3 adapter for caching
- created common package with types and aws utilities

### CODE REVIEW

File: packages/s3-cache/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { S3 } from '@aws-sdk/client-s3'
```
Comment (asa-dbb) [inline packages/s3-cache/src/index.ts:1]: Would it makes sense to use our wrapper? https://github.com/DBB-Software/platform/blob/main/packages/common/src/aws/s3.ts
Or at least logger aws-xray-sdk-core

File: packages/s3-cache/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import { getAWSCredentials, type CacheEntry, type CacheStrategy } from '@dbbs/next-cache-handler-common'
+
+export class S3Cache implements CacheStrategy {
```
Comment (asa-dbb) [inline packages/s3-cache/src/index.ts:4]: Does it make sense to use extra-memory cache here? 
The one from memory strategy or 
https://github.com/DBB-Software/platform/blob/main/packages/common/src/domain/setting-service-client.ts
for mainly de-bouncing

File: packages/cache-core/src/cacheHandler.ts [hunk 3]
```diff
@@ -97,15 +104,15 @@ export class Cache implements CacheHandler {
...
-      if (Cache.noCacheRoutes.includes(cacheKey)) {
+      if (Cache.noCacheRoutes.includes(pageKey)) {
         return null
       }
 
-      Cache.logger.info(`Reading cache data for ${cacheKey}`)
+      Cache.logger.info(`Reading cache data for ${pageKey}`)
 
-      const data = await Cache.cache.get(this.getPageCacheKey(cacheKey), {
+      const data = await Cache.cache.get(pageKey, this.getPageCacheKey(pageKey), {
```
Comment (asa-dbb) [inline packages/cache-core/src/cacheHandler.ts:115]: Or maybe here it makes sense to add cache

Comment (RomanBobrovskiy) [reply packages/cache-core/src/cacheHandler.ts:115]: this is a core functionality which should not be altered, you can define just what cache strategy you want to use (like, memory, redis, s3, .etc)

File: packages/s3-cache/src/index.ts (hunk 1 above)
Comment (RomanBobrovskiy) [reply packages/s3-cache/src/index.ts:1]: this has to be a published package https://github.com/DBB-Software/platform/blob/main/packages/common/src/aws/s3.ts, + we really need 2 methods from s3 library at this moment

File: packages/s3-cache/src/index.ts (hunk 2 above)
Comment (RomanBobrovskiy) [reply packages/s3-cache/src/index.ts:4]: not sure that it needs to be there, it is more like hybrid strategy already, but i am thinking that we can make it as flag for s3 strategy to extra level of caching. But again. based on the project's size memory cache won't be really proactive at this point

**Review by asa-dbb — APPROVED** [review]

**Pull request merged**